File: 프로그래머스/lv2/42842. 카펫/카펫.py
def solution(brown, yellow):
    answer = []
    total = brown + yellow
    
    height=3
    width=3 #최소 3x3은 되어야함
    
    while(1):
        if total % height != 0: #높이랑 안나누어떨어지면 정수 될때까지 나눠
            height += 1
            continue
        else: #나누어떨어지면 width랑 height를 정해보자
            width = total / height
        
        if (width-2)*(height-2) != yellow: #넓-2 x 높-2 가 옐로면 정답인거고 아니면 높이 추가해보자
            height += 1
        else :
            answer.append(width)
            answer.append(height)
            print(answer)
            return answer

File: 프로그래머스/lv2/42842. 카펫/카펫.js
function solution(brown, yellow) {
    var answer = [];
    
    let total = brown + yellow
    
    for (let i = 3; i<Math.sqrt(total)+1; i++) {
        
        if (total % i === 0) {
            let x = total / i
            if(((x-2)*(i-2) === yellow) && (x*i === total)  ) {
                answer.push(x)
                answer.push(i)
                return answer
            }
        }
    }
    // (x-2)*(y-2) = yellow
    // x*y = brown + yellow
    
    return answer;
}